Add basic tests

nmattia opened the pull request:
I'm doing some work on a separate branch and I don't want to break anything. @dackerman have I missed a functionality?

dackerman commented on test/Main.hs:
```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE QuasiQuotes #-}
+
```
we need `{-# LANGUAGE OverloadedStrings #-}` here to get the literals to compile

dackerman commented on test/Main.hs:
```diff
@@ -0,0 +1,33 @@
+anti1 = [jsx|<div>{Dom.text "Hello"}</div>|]
+anti2 = [jsx|<div>{Dom.text "Hello"}{Dom.text "World"}</div>|]
+anti3 = let str = "Hello" in [jsx|<div>{Dom.text str}</div>|]
+anti4 = [jsx|<div style={"color: " ++ color ++ ";width:10px;height:10px"}/>|]
```
I think you need `Data.Text.concat` instead of the string operations

dackerman commented on test/Main.hs:
```diff
@@ -0,0 +1,33 @@
+
+-- These test antiquotation (MonadWidget values + dynamic attributes)
+anti1, anti2, anti3, anti4, anti5 :: (MonadWidget t m) => m ()
+anti1 = [jsx|<div>{Dom.text "Hello"}</div>|]
```
If you want, you could try adding a nested `<div>` with quasiquotes inside. This will be especially cool to test when we have values being returned from within the jsx block!

dackerman commented:
Hi @nmattia, this is fantastic! Thanks for adding some tests. I think we can add some more edge cases for the parser and quasiquoter at some point, but this is a great start!

I did try to run this with `cabal configure --enable-tests --ghcjs && cabal test` and ran into a few snags which I commented on. Also, can you add the test running command to the `.travis.yml` file? Currently it just runs the build.

dackerman commented:
I'm realizing from your email that I was running on a later reflex version when testing this, hence my Text suggestions. So I guess you're right to use Strings here, but could you update the cabal file with bounds?

nmattia commented:
I've updated the stack file to use GHC 8 and updated the code accordingly. I've also added bounds on the cabal though the code as it is now should build with both `String` and `Text`.

@dackerman For travis: would you rather build the project with nix or stack? Ideally we'd make sure it builds with both.

dackerman commented:
@nmattia if you want to add stack build/test to travis as well, that works for me, I just want to have the tests being run in some way.

nmattia commented:
Stack build on travis is being very picky, I'm giving up for now.

dackerman commented:
Yeah travis is annoying to set up - really long feedback cycle. We can worry about it later - this looks good for now!
